MSProject/ViewControllers/LibraryTableView/WBLibraryTableViewController.swift

```diff
@@ -46,49 +41,51 @@ class WBLibraryTableViewController: UIViewController {
         
     }
 
-    // MARK: - Private
-    private func initLibraryTableViewModel() {
-       
-        libraryViewModel.showAlertClosure = { [weak self] (error) in
-            MBProgressHUD.hide(for: self?._view, animated: true)
-            DispatchQueue.main.async {
-                self?.showAlertMessage(message: error.localizedDescription)
-            }
-        }
-        
-        libraryViewModel.reloadViewClosure = { [weak self] () in
-            MBProgressHUD.hide(for: self?._view, animated: true)
-            DispatchQueue.main.async {
-                self?._view.bookTable.reloadData()
-            }
-        }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         
-        MBProgressHUD.showAdded(to: _view, animated: true)
-        libraryViewModel.loadBooks()
+        loadBooks()
     }
     
+    // MARK: - Private
     private func configureTableView() {
         _view.bookTable.delegate = self
         _view.bookTable.dataSource = self
         
+        let nib = UINib.init(nibName: "WBBookTableViewCell", bundle: nil)
+        _view.bookTable.register(nib, forCellReuseIdentifier: "WBBookTableViewCell")
+        
         _view.configureLibraryTableView()
     }
     
     // MARK: - Services
     @objc private func loadBooks() {
         MBProgressHUD.showAdded(to: _view, animated: true)
-        libraryViewModel.loadBooks()
-        DispatchQueue.main.async {
-            self._view.bookTable.refreshControl?.endRefreshing()
+        viewModel.repository.getBooks().startWithResult { [unowned self] result in
+            switch result {
+            case .success(let value):
+                self._view.bookTable.refreshControl?.endRefreshing()
+                MBProgressHUD.hide(for: self._view, animated: true)
+                self.loadTableWithBooks(books: value)
```

Comment:
you can use the on(value: ) operator in the getBooks() function to update the viewmodel value, so you don't need to call the viewmodel again in the closure.